src/Authorization.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+namespace Minishlink\WebPush;
+
+use Base64Url\Base64Url;
+use InvalidArgumentException;
+
+class Authorization implements Contracts\AuthorizationInterface
+{
+    /**
+     * @var string
+     */
+    private $privateKey;
+    /**
+     * @var string
+     */
+    private $publicKey;
+    /**
+     * @var string
+     */
+    private $subject;
+
+    /**
+     * @param string $private_key
+     * @param string $public_key
+     * @param string $subject
+     *
+     * @throws InvalidArgumentException
+     */
+    public function __construct(string $private_key, string $public_key, string $subject)
```

Comment:
It looks like the setters have a lot of complexity.
I am not sure to remember why the keys are base 64 encoded, but it will be easier if binary strings are passed as arguments.

Reply:
Awesome, I'd prefer that as well, will make this change - was following suit of the `VAPID::validate()` call

Reply:
On second thought, I think they may actually be best passed base64 encoded due to the fact that they’ll need to also likely be stored as binary data. Otherwise we’d be expecting consumers to store as base64 and then decode the values before using them. 

https://stackoverflow.com/questions/10979100/how-to-store-binary-data-in-mysql

Let me know what you think.